crates/hdk/src/link/link_types.rs

```diff
@@ -0,0 +1,75 @@
+use hdi::prelude::LinkTypeFilterExt;
+use holo_hash::{ActionHash, AgentPubKey, AnyLinkableHash};
+use holochain_wasmer_guest::WasmError;
+use holochain_zome_types::{GetLinksInput, LinkTag, Timestamp};
+
+/// A builder to streamline creating a `GetLinksInput`
+#[derive(PartialEq, Clone, Debug)]
+pub struct GetLinksInputBuilder(GetLinksInput);
```

Comment:
This newtype is a bit of a fiddle. It replaces the implementation of `GetLinksInput::new` with `GetLinksInputBuilder::try_new` so that we can access `LinkTypeFilterExt` which was being relied upon for calls to `get_links` when that took 3 parameters instead of `GetLinksInput`.

I couldn't find a cleaner way to lay out the types because `LinkTypeFilterExt` depends on `zome_info` so it has to live in either the hdi or hdk. It's used in the hdi, so that makes that choice. Then similarly for `GetLinksInput` it's referenced by zome_io so it can't leave the holochain_zome_types.